oci/oci_dest.go

```diff
@@ -107,17 +109,10 @@ func (d *ociImageDestination) PutManifest(m []byte) error {
 // to any other readers for download using the supplied digest.
 // If stream.Read() at any time, ESPECIALLY at end of input, returns an error, PutBlob MUST 1) fail, and 2) delete any data stored so far.
 // Note: Calling PutBlob() and other methods may have ordering dependencies WRT other methods of this type. FIXME: Figure out and document.
-func (d *ociImageDestination) PutBlob(digest string, stream io.Reader) error {
-	blobPath, err := d.ref.blobPath(digest)
+func (d *ociImageDestination) PutBlob(stream io.Reader) (string, int64, error) {
+	blobFile, err := ioutil.TempFile("", "oci-put-blob")
```

Comment:
The temporary file should really be created in the destination directory; the default is `/tmp`, which is tempfs, and cross-filesystem `rename` usually fails.